### PR TITLE
DOC: remove warnings in doc build

### PR DESCRIPTION
**Pull request by FranzForstmayr**
#### Reference issue
Another approach to fix CI as alternatives to #18471 and #18472

#### What does this implement/fix?
Some small fixes in docstrings to avoid warnings.

### CODE REVIEW

**Comment by FranzForstmayr** [comment]
For some reason I'm still unable to reproduce the doc build failure from #15755 locally, let's see what CI does.

**Comment by FranzForstmayr** [comment]
@h-vetinari 
There are still some reported warnings in the sphinx build I haven't fixed now. 

One example is https://github.com/scipy/scipy/blob/main/scipy/stats/_stats_py.py#L1692 which even mentions that a warning is thrown. How should we handle this cases?

**Comment by mdhaber** [comment]
To get CI green, it's OK with me to revert the changes to these examples made in gh-17765. With the introduction of notebook tutorials, we plan to move these detailed examples to tutorials, anyway. CC @tupui.

**Comment by FranzForstmayr** [comment]
Still having issues with cosine and bohman:
https://output.circle-artifacts.com/output/job/e748756c-73da-4977-a466-32c21b2b119e/artifacts/0/html/reference/generated/scipy.signal.windows.bohman.html#scipy.signal.windows.bohman
https://output.circle-artifacts.com/output/job/e748756c-73da-4977-a466-32c21b2b119e/artifacts/0/html/reference/generated/scipy.signal.windows.cosine.html#scipy.signal.windows.cosine

seems strange to me, as these weren't touched since a while.
Any ideas?

If I test the example code locally I don't get any warning.

**Comment by h-vetinari** [comment]
One thing that I noted in the logs of all the failing runs (and not present in the passing ones) are the following weird-looking warnings:
```
reading sources... [ 10%] reference/generated/scipy.integrate.ode.set_integrator .. reference/generated/scipy.interpolate.LinearNDInterpolator.__call__
<string>:19: DeprecationWarning: invalid escape sequence \s
reading sources... [ 13%] reference/generated/scipy.interpolate.NdPPoly .. reference/generated/scipy.io.mminfo
reading sources... [ 16%] reference/generated/scipy.io.mmread .. reference/generated/scipy.linalg.blas.zher
reading sources... [ 20%] reference/generated/scipy.linalg.blas.zher2 .. reference/generated/scipy.linalg.lapack.chesv_lwork
reading sources... [ 23%] reference/generated/scipy.linalg.lapack.chesvx .. reference/generated/scipy.linalg.lapack.dpbtrf
<string>:50: DeprecationWarning: invalid escape sequence \d
<string>:46: DeprecationWarning: invalid escape sequence \l
reading sources... [ 26%] reference/generated/scipy.linalg.lapack.dpbtrs .. reference/generated/scipy.linalg.lapack.sormrz
<string>:46: DeprecationWarning: invalid escape sequence \,
<string>:46: DeprecationWarning: invalid escape sequence \,
<string>:50: DeprecationWarning: invalid escape sequence \d
<string>:30: DeprecationWarning: invalid escape sequence \i
<string>:25: DeprecationWarning: invalid escape sequence \i
<string>:25: DeprecationWarning: invalid escape sequence \i
<string>:42: DeprecationWarning: invalid escape sequence \c
<string>:42: DeprecationWarning: invalid escape sequence \c
<string>:22: DeprecationWarning: invalid escape sequence \P
reading sources... [ 30%] reference/generated/scipy.linalg.lapack.sormrz_lwork .. reference/generated/scipy.linalg.lapack.zhegvd
```

Unfortunately it's not clear at all to me from which docstring they come from.

> Still having issues with cosine and bohman

Aside from that very tenuous lead, I have no idea what's affecting the bohman plot. Even more mysteriously, the cosine one is also [missing](https://output.circle-artifacts.com/output/job/e748756c-73da-4977-a466-32c21b2b119e/artifacts/0/html/reference/generated/scipy.signal.windows.cosine.html) for #18472, but [present](https://output.circle-artifacts.com/output/job/6c7795f5-50ac-48fd-ae6c-f7251d6ca4e1/artifacts/0/html/reference/generated/scipy.signal.windows.cosine.html) for #18471

**Comment by FranzForstmayr** [comment]
I can see the deprecation warnings in my local build too, but doesn't fail doc build.

There's some really weird caching somewhere.

File: doc/source/conf.py [hunk 1]
```diff
@@ -319,6 +319,7 @@
         'lsim2 is deprecated',  # Deprecation of scipy.signal.lsim2
         'impulse2 is deprecated',  # Deprecation of scipy.signal.impulse2
         'step2 is deprecated',  # Deprecation of scipy.signal.step2
+        'interp2d` is deprecated',  # Deprecation of scipy.interpolate.interp2d
```
Comment (WarrenWeckesser) [inline doc/source/conf.py:322]: That backtick after `interp2d` shouldn't be there.

Comment (h-vetinari) [reply doc/source/conf.py:322]:
```suggestion
        'interp2d is deprecated',  # Deprecation of scipy.interpolate.interp2d
```

Comment (h-vetinari) [reply doc/source/conf.py:322]: Well, that backtick is in the message of the deprecation warning. However, something really strange is going on also with the rendering in the GH UI - the first line is basically invisible:

interp2_dep_I
except when highlighted
interp2_dep_II

Comment (h-vetinari) [reply doc/source/conf.py:322]: I think it might be easiest to remove the backtick from `both` the deprecation message and the warning filter.

Comment (h-vetinari) [reply doc/source/conf.py]: Ah, the backticks are actually inserted by `@np.deprecate` 🤦

**Comment by h-vetinari** [comment]
@FranzForstmayr, sorry for force-pushing your branch, but it was conflicting after #18472 got merged. I added a revert for that and based your changes on top. Feel free to rebase & force-push again if you want.

File: doc/source/conf.py (hunk 1 above)
Comment (h-vetinari) [reply doc/source/conf.py:322]: Github rendering aside, the backtick belongs in that error message, so I'm resolving this.

**Review by mdhaber — APPROVED** [review]
I approve the doc changes to `normaltest` and `kurtosistest`. This PR makes them approximately as they were before gh-17765. @tupui just wanted to make sure this was on your radar. When we turn the examples into notebooks, we should also review these examples.

**Review by mdhaber — DISMISSED** [review]
I approve the doc changes to `normaltest` and `kurtosistest`. This PR makes them approximately as they were before gh-17765. @tupui just wanted to make sure this was on your radar. When we turn the examples into notebooks, we should also review these examples.

**Comment by h-vetinari** [comment]
> I approve the doc changes to `normaltest` and `kurtosistest`. This PR makes them approximately as they were before [gh-17765](https://github.com/scipy/scipy/pull/17765)

Normally I wouldn't advocate the removal of something useful, but since you suggested early on in this thread to revert those, I didn't look more closely at that. Now that we don't have the urgency or broken CI, should we take the time to remove the reverts and fix it properly? Or is that not worth the effort IYO?

**Comment by mdhaber** [comment]
If it's easy to revert the reverts, that would be best. I'm comfortable either way, though.

**Comment by h-vetinari** [comment]
```
    warnings.warn("kurtosistest only valid for n>=20 ... continuing "
UserWarning: kurtosistest only valid for n>=20 ... continuing anyway, n=11
```

Can we increase `n` in the example?

**Comment by mdhaber** [comment]
This was part of the point of the example - to show what happens when the number of observations is too small.

**Comment by h-vetinari** [comment]
> I know it's ugly, but suppress the warning for a quick fix?

If it's the intention to showcase the "bad" case, I don't think suppressing the warning is bad per se

**Comment by h-vetinari** [comment]
@mdhaber @j-bowhay, any remaining comments here? Aside from reinstating the lsmi2 etc. doc string examples, it's now really only very small changes to fix the currently existing warnings in the sphinx build.

**Review by j-bowhay — APPROVED** [review]

**Review by mdhaber — APPROVED** [review]
Sure. It looks like all that was needed in stats was to add the `kurtosistest` to a list of warnings? That's great.

**Comment by j-bowhay** [comment]
Thanks everyone

**Pull request merged**